src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/HtmlMarkupParser.cs

```diff
@@ -1357,8 +1357,6 @@ private void ParseJavascriptAndEndScriptTag(in SyntaxListBuilder<RazorSyntaxNode
                 var tagStart = CurrentStart;
                 builder.Add(OutputAsMarkupLiteral());
 
-                SpanContext.EditHandler.AcceptedCharacters = endTagAcceptedCharacters;
```

Comment:
Why'd this have to move?

Reply:
So that the correct accepted characters gets set for the end tag (when GetNodeWithSpanContext is called in line 1402). I'll add a test.